tests/run/genexpr_arg_order.py

```diff
@@ -0,0 +1,178 @@
+# mode: run
+# tag: genexpr, py3, py2
+
+from __future__ import print_function
+
+# Tests that function arguments to generator expressions are
+# evaluated in the correct order (even after optimization)
+
+import cython
+
+@cython.cfunc
+@cython.returns(cython.int)
+def f():
```

Comment:
Why not call these functions `zero()`, `five()` and `one()` ? Makes the tests easier to understand.
Or `get_0()`, `get_5()` and `get_1()`, if you prefer.
And why print 1, 2, 3 instead of the function name?
(Ultimately, the goal of test output is to make it obvious _what_ went wrong when they fail. The less indirection and manual code tracing, the better.)